### ISSUE TITLE
[🐛 Bug]: The class locator combined with Partial Link Text is returning error for nested elements in wdio v8

### ISSUE THREAD

**Issue by blackforest-t** [webdriverio/webdriverio#11154]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

8.16.3

### Node.js Version

16.19.0

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
{
browserName: chrome,
browserVersion: 116
}
```

### What happened?

I have selectors based on the partial text, it should find me the element in which the text is.
For example, I have the following HTML:

```HTM
  <div role="tab" class="mdc-tab mat-mdc-tab">
    <span class="mdc-tab__ripple"></span>
    <div class="mat-ripple mat-mdc-tab-ripple"></div>
    <span class="mdc-tab__content">
      <span class="mdc-tab__text-label">TextToFind sample</span>
    </span>
    <span class="mdc-tab-indicator">
      <span class="mdc-tab-indicator__content" style=""></span>
    </span>
  </div>
```
  
And I have a selector to select this whole element: `.mat-mdc-tab*=TextToFind`
  
By using browser.$(<selector>) in the wdio v7, that selector gave me exactly the `<div role="tab" class="mdc-tab mat-mdc-tab">...</div>` element, and now (in v8) it gives me error (no element found). 
  
I've noticed such behavior only with class selectors combined with Link/Partial link.
But if I change, for example, the class selector to the attribute selector, like this `[role="tab"]*=TextToFind`
Then it works fine and gives exactly what I need (and what I had in v7).

### What is your expected behavior?

To sum it up:
* selector `.mat-mdc-tab*=TextToFind` returns error, no element found.
* selector `[role="tab"]*=TextToFind` returns the whole element with nested elements where text has been found.

Which is not expected behavior for me. I'm expecting that the logic would be the same as it was in v7. I'm expecting the locator  `.mat-mdc-tab*=TextToFind` to return me the whole element with nested elements where text has been found.

### How to reproduce the bug.

Create an HTML file:
```HTM
  <div role="tab" class="mdc-tab mat-mdc-tab">
    <span class="mdc-tab__ripple"></span>
    <div class="mat-ripple mat-mdc-tab-ripple"></div>
    <span class="mdc-tab__content">
      <span class="mdc-tab__text-label">TextToFind sample</span>
    </span>
    <span class="mdc-tab-indicator">
      <span class="mdc-tab-indicator__content" style=""></span>
    </span>
  </div>
```

Try to find element by using the REPL interface:
`browser.$('.mat-mdc-tab*=TextToFind').getHTML()` 

Those actions should return HTML like: `<div role="tab" class="mdc-tab mat-mdc-tab">...</div>` but there's error instead

### Relevant log output

```typescript
N/A
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by erwinheitzman** [comment]
This is expected behavior and there's nothing wrong with making breaking changes between major versions.
I can't remember having this in the release notes of v8 though so we can improve on our communication and/or release strategy.

**Comment by blackforest-t** [comment]
> This is expected behavior and there's nothing wrong with making breaking changes between major versions. I can't remember having this in the release notes of v8 though so we can improve on our communication and/or release strategy.

I've noticed a few bugs with similar root causes. But I'm not sure if this one is related.

https://github.com/webdriverio/webdriverio/pull/9992
https://github.com/webdriverio/webdriverio/issues/10040
https://github.com/webdriverio/webdriverio/issues/10067

**Comment by blackforest-t** [comment]
In addition. Link/partial link combined with attributes works fine (e.g: `[role="columnheader"]*=TextToFind`), but Link/partial link combined with classes - not

**Comment by christian-bromann** [comment]
@blackforest-t sorry but with this HTML blob it is very difficult to understand what you are looking for. Mind providing a reproducible example that clarifies this? Thank you!

**Comment by blackforest-t** [comment]
> @blackforest-t sorry but with this HTML blob it is very difficult to understand what you are looking for. Mind providing a reproducible example that clarifies this? Thank you!

@christian-bromann 
My bad. Sorry. Updated with a simplier example. 
Thanks

**Comment by erwinheitzman** [comment]
Ah from your previous example I thought the issue was that you expected a different element with the selector you used.

I see now, I can test this some time soon 👍

**Comment by blackforest-t** [comment]
> Ah from your previous example I thought the issue was that you expected a different element with the selector you used.
> 
> I see now, I can test this some time soon 👍

Correct. But better start with this.

**Comment by erwinheitzman** [comment]
I can verify that this is an issue. Seems like the element can be found but it only looks at the first class when an element has multiple classes.

**Comment by erwinheitzman** [comment]
I tested some more and the cause is this line: https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/src/utils/findStrategy.ts#L267

In the example given, the full selector will be:
```
.//[contains(@class, "mat-mdc-tab") and contains(., "TextToFind") and not(.//[contains(@class, "mat-mdc-tab")])]
```
Which translates to find the element with the class mat-mdc-tab and text TextToFind but it cannot have class mat-mdc-tab. @christian-bromann I noticed you specifically made this change in #10130 and I am wondering if we can remove this line or if it is needed?

**Comment by erwinheitzman** [comment]
That said, this will also find elements that contain mat-mdc-tab, so it will also find foo-mat-mdc-tab-bar.

If we would like to improve this so that it only finds the matching class we should change it to: `contains(concat(' ',@class,' '), ' mat-mdc-bar ')`

**Comment by erwinheitzman** [comment]
So after some more digging I finally found the EXACT cause.

When we have this html snippet:
```
  <div role="tab" class="mdc-tab mat-mdc-tab">
    <span class="mdc-tab__ripple"></span>
    <div class="mat-ripple mat-mdc-tab-ripple"></div>
    <span class="mdc-tab__content">
      <span class="mdc-tab__text-label">TextToFind sample</span>
    </span>
    <span class="mdc-tab-indicator">
      <span class="mdc-tab-indicator__content" style=""></span>
    </span>
  </div>
```

And we execute the following:
```js
await $('.mdc-tab*=TextToFind').getHTML()
```

It will create the following selector under the hood:
```
.//*[contains(@class, "mat-mdc-tab") and contains(., "TextToFind") and not(.//*[contains(@class, "mat-mdc-tab")])]
```

To explain what happens, I will take a html snippet from one of our component tests (Lit components):
```
<div>
  <div>
    <div>
      Find me
    </div>
  </div>
</div>
```

Now when querying `.//div[normalize-space(text()) = "Find me"]` (in this example it does not matter if it is partial or not) xpath actually does not return `<div>Find me</div>` but will return the first div that it meets which will be:
```
<div>
  <div>
    <div>
      Find me
    </div>
  </div>
</div>
```

Now the interesting part comes.
The ` and not(.//*[contains(@class, "mat-mdc-tab")])` makes sure that instead of the first element, we get the last (the before mentioned `<div>Find me</div>`).

Now all of the selectors do actually work, so why does it fail for the snippet provided by @blackforest-t ?

Turns out that because the `and not` condition states (simplified) `and not where element contains class mat-mdc-tab`.
Now is there an element in the DOM tree that matches this criteria? Well YES!
```
<div class="mat-ripple mat-mdc-tab-ripple"></div>
```
This is the element that is causing the issue as this meets the criteria.

If you remove that element from the DOM, the selector works without issues.

Should be fixed by #11206
